src/diff_edit.rs

```diff
@@ -48,6 +48,8 @@ pub enum DiffEditError {
     },
     #[error("I/O error: {0:?}")]
     IoError(#[source] std::io::Error),
+    #[error("Failed to snapshot changes: {0:?}")]
+    SnapshotError(SnapshotError),
```

Comment:
```suggestion
    SnapshotError(#[from] SnapshotError),
```

Maybe we should define this as `#[from]`? In principle, that would preserve the error chain better, I think.